Achievements/SelfMade.lua

```diff
@@ -57,35 +55,34 @@ local function isSelfCreated(...)
                     return true
                 end
             end
-            if i >= 18 then -- this ensures that the loop ends
-                if text == nil then
-                    GameTooltip:Hide() --prevents a hung empty tooltip window
-                    return false
-                end
-            end
         end
     end
-
+    GameTooltip:Hide() --prevents a hung empty tooltip window
+    return false
 end
 
 -- Register Definitions
 self_made_achievement:SetScript("OnEvent", function(self, event, ...)
     GameTooltip:SetOwner(WorldFrame, "ANCHOR_CURSOR") -- Anchors the cursor so it ensure we check the right tooltip
     local arg = { ... }
     if event == "PLAYER_EQUIPMENT_CHANGED" then
-        if arg[2] == true then -- this are is checking if the equippable spot is nil
+        if arg[2] == true then -- this arg[2] is checking if the equippable spot is nil
+            return
+        end
+        if arg[1] == 19 then -- This is checking if what you have equipped is a tabard, we cant make those :D
+            print("You put on a tabard! You're Stylish.")
             return
         end
     elseif event == "ITEM_UNLOCKED" and arg[2] == nil then -- need to ensure that arg[2] is nil because it is only nil when a bag is being put into the bag "CheckBox" on the action bar.
         local item_id = GetInventoryItemID("player", arg[1])
         local item_name, _, _, _, _, _, item_subtype = GetItemInfo(item_id)
         GameTooltip:SetInventoryItem("player", arg[1]) -- this arg[1] passes the invSlot to  be checked.
-	-- Should these checks fail, the player fails the achievement.
+        -- Should these checks fail, the player fails the achievement.
         if isSelfCreated(GameTooltip:GetRegions(arg[1])) == false then
             if item_subtype ~= "Fishing Poles" then
                 if Start(arg[1]) == false then
                     Hardcore:Print("Equipped " .. item_name .. " which isn't self created.")
-                    self_made_achievement.fail_function_executor.Fail(self_made_achievement.name)
+                    --self_made_achievement.fail_function_executor.Fail(self_made_achievement.name)
```

Comment:
Did you intend to comment this line?

Reply:
that was for testing purposes it can be uncommented, knew i missed something